vortex-array/src/scalar/mod.rs

```diff
@@ -99,42 +105,41 @@ impl Scalar {
     }
 
     pub fn null(dtype: &DType) -> Self {
+        assert!(dtype.is_nullable());
         match dtype {
             DType::Null => NullScalar::new().into(),
-            DType::Bool(_) => BoolScalar::new(None).into(),
+            DType::Bool(_) => BoolScalar::none().into(),
             DType::Int(w, s, _) => match (w, s) {
                 (IntWidth::Unknown, Signedness::Unknown | Signedness::Signed) => {
-                    PrimitiveScalar::none(PType::I64).into()
+                    PrimitiveScalar::none::<i64>().into()
                 }
                 (IntWidth::_8, Signedness::Unknown | Signedness::Signed) => {
-                    PrimitiveScalar::none(PType::I8).into()
+                    PrimitiveScalar::none::<i8>().into()
                 }
                 (IntWidth::_16, Signedness::Unknown | Signedness::Signed) => {
-                    PrimitiveScalar::none(PType::I16).into()
+                    PrimitiveScalar::none::<i16>().into()
                 }
                 (IntWidth::_32, Signedness::Unknown | Signedness::Signed) => {
-                    PrimitiveScalar::none(PType::I32).into()
+                    PrimitiveScalar::none::<i32>().into()
                 }
                 (IntWidth::_64, Signedness::Unknown | Signedness::Signed) => {
-                    PrimitiveScalar::none(PType::I64).into()
-                }
-                (IntWidth::Unknown, Signedness::Unsigned) => {
-                    PrimitiveScalar::none(PType::U64).into()
+                    PrimitiveScalar::none::<i64>().into()
                 }
-                (IntWidth::_8, Signedness::Unsigned) => PrimitiveScalar::none(PType::U8).into(),
-                (IntWidth::_16, Signedness::Unsigned) => PrimitiveScalar::none(PType::U16).into(),
-                (IntWidth::_32, Signedness::Unsigned) => PrimitiveScalar::none(PType::U32).into(),
-                (IntWidth::_64, Signedness::Unsigned) => PrimitiveScalar::none(PType::U64).into(),
+                (IntWidth::Unknown, Signedness::Unsigned) => PrimitiveScalar::none::<u64>().into(),
```

Comment:
nit: might be more readable/consistent if we consolidate `Unknown` width with the _64 branch by making it `(IntWidth::_64 | IntWidth::Unknown, Signedness::Unsigned). Ditto for signed integer and for float.

Reply:
I'm about to get rid of Unknown anyways